torch/_inductor/kernel/flex_attention.py

```diff
@@ -228,25 +228,35 @@ def build_subgraph_buffer(
     k_offset = off_z * stride_kz + off_h * stride_kh
     v_offset = off_z * stride_vz + off_h * stride_vh
 
-    SPARSE_Z = {{size("SPARSE_KV_NUM_BLKS", 0)}}
-    SPARSE_H = {{size("SPARSE_KV_NUM_BLKS", 1)}}
+    use_sparse_mask: tl.constexpr = (not SPARSE_Q_BLOCK_SIZE == 0) and (not SPARSE_KV_BLOCK_SIZE == 0)
+    if use_sparse_mask:
```

Comment:
Out of curiosity, any particular reason you added this?

Reply:
This allows an empty mask to have SPARSE_BLOCK_SIZE of 0 instead of seq_len. This allows seq_len to be dynamic, while SPARSE_BLOCK_SIZE remain static.

Reply:
Reverted specific path for no mask. Fallback a full mask of BLOCK 128x128 when no block_mask is specified.